### PR TITLE
partial: exception cleanup and add todos

### PR DESCRIPTION
**Pull request by aakropotkin**
- Adds real exception handling to manifest/lockfile parsing.
- Adds TODOs to some Manifest and Lockfile routines to delegate work.

### CODE REVIEW

**Comment by stahnma** [comment]
Please attach to an issue and put in sprint and give an estimate.

File: include/flox/package.hh [hunk 1]
```diff
@@ -170,42 +170,47 @@ public:
...
-    if ( os.has_value() ) { j[system].emplace( "version", *os ); }
-    else { j[system].emplace( "version", nlohmann::json() ); }
+    nlohmann::json jto = { { system,
+                             { { "name", this->getFullName() },
+                               { "pname", this->getPname() } } } };
 
-    os = this->getSemver();
-    if ( os.has_value() ) { j[system].emplace( "semver", *os ); }
-    else { j[system].emplace( "semver", nlohmann::json() ); }
+    std::optional<std::string> oos = this->getVersion();
```
Comment (zmitchell) [inline include/flox/package.hh:179]: Nitpick: These really short variable names are hard to read. I have no idea what `oos` means.

**Review by zmitchell — APPROVED** [review]

File: include/flox/package.hh (hunk 1 above)
Comment (aakropotkin) [reply include/flox/package.hh:179]: It was just `os` before ;)
I made it 3 characters to get `clang-tidy` to stop griping.

**Review by garbas — APPROVED** [review]

**Comment by garbas** [comment]
Waiting for this to rebase my work for https://github.com/flox/product/issues/551 on top of it!

**Pull request merged**